File: src/main/java/com/nofront/plantinggrassbe/domain/UserSave.java
package com.nofront.plantinggrassbe.domain;

import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicInsert;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.time.LocalDateTime;

@Getter
@DynamicInsert
@Entity
@Data
@Builder
@EntityListeners(AuditingEntityListener.class)
public class UserSave {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

//    @ManyToOne(targetEntity = User.class)
//    @JoinColumn(name = "username")
    private String username;


    @Column()
    boolean isPrivate;

    @CreatedDate
    @Column()
    LocalDateTime createdDate;

    @LastModifiedDate
    @Column()
    LocalDateTime updatedDate;

    private String data;

    private String dataKey;

}

File: src/main/kotlin/com/dongholab/usersave/entity/UserSave.kt
package com.dongholab.usersave.entity

import com.dongholab.usersave.r2dbc.R2dbcPersistable
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Table("user_save")
data class UserSave (
    @Column("user_id")
    val userId: String,
    @Column("data_key")
    val dataKey: String?,
    @Column("data")
    val data: String?,
    @Column("private")
    val private: Boolean,
    @Column("created_date")
    var createdDate: LocalDateTime
): R2dbcPersistable<Long>() {
    @Id
    @Column("id")
    var id: Long? = null
    constructor(id: Long, userId: String, dataKey: String, data: String, private: Boolean, currentDateTime: LocalDateTime): this(userId, dataKey, data, private, currentDateTime) {
        this.id = id
        this.setId(id)
    }
}